woocommerce/google-listings-and-ads
# Disabled toggle with tooltip for free listings in Programs table

### Changes proposed in this Pull Request:



Closes #487 .

This PR displays a disabled ToggleControl for Free Listings in the Programs table.

#### Technical note:

Note that to achieve this, I have upgraded our `@wordpress/components` package to the version 12.0.8 (see https://github.com/woocommerce/google-listings-and-ads/pull/492#issuecomment-822015577). The ToggleControl just recently supported `disabled` in early 2021 (see PR https://github.com/WordPress/gutenberg/pull/28228). Unfortunately there isn't a real quick way to find out the earliest version in which the change is available.

Also, I have created an `AppTooltip` component and made a little refactor on `MetricNumber` using this new component. The `AppTooltip` component makes the Tooltip display and wrap around the ToggleControl better.

### Screenshots:

![free listings disabled toggle with tooltip](https://user-images.githubusercontent.com/417342/115018220-2ed24480-9eea-11eb-9d55-c4441d979f2e.gif)

### Detailed test instructions:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard
2. Hover your mouse over the disabled Free Listings toggle in the Enabled column in Programs table.
3. There should be a tooltip displayed.
4. Move your mouse away from the toggle. The tooltip should disappear.

### Changelog Note:

Display disabled toggle with tooltip for free listings in Programs table.
